### PR TITLE
Implement FromIter for &u32, &u64

### LINKED ISSUE #201: Impl FromIter for &u32
For ergonomics mostly. Likewise with &u64 on treemap

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### Implement FromIter for &u64 for treemap

Fixes #201.

### CODE REVIEW

File: src/bitmap/iter.rs [hunk 1]
```diff
@@ -139,6 +139,14 @@ impl FromIterator<u32> for RoaringBitmap {
     }
 }
 
+impl<'a> FromIterator<&'a u32> for RoaringBitmap {
+    fn from_iter<I: IntoIterator<Item = &'a u32>>(iterator: I) -> RoaringBitmap {
```
Comment (Kerollmops) [inline src/bitmap/iter.rs:143]: Is it really necessary to specify the lifetimes here?

```suggestion
impl FromIterator<&u32> for RoaringBitmap {
    fn from_iter<I: IntoIterator<Item = &u32>>(iterator: I) -> RoaringBitmap {
```

File: src/bitmap/iter.rs [hunk 2]
```diff
@@ -147,6 +155,14 @@ impl Extend<u32> for RoaringBitmap {
     }
 }
 
+impl<'a> Extend<&'a u32> for RoaringBitmap {
+    fn extend<I: IntoIterator<Item = &'a u32>>(&mut self, iterator: I) {
```
Comment (Kerollmops) [inline src/bitmap/iter.rs:159]:
```suggestion
impl Extend<&u32> for RoaringBitmap {
    fn extend<I: IntoIterator<Item = &u32>>(&mut self, iterator: I) {
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! I just have some concerns about lifetimes but overall it looks great!

File: src/bitmap/iter.rs (hunk 1 above)
Comment (RaduBerinde) [reply src/bitmap/iter.rs:143]: It doesn't work, I get:
```
error[E0637]: `&` without an explicit lifetime name cannot be used here
   --> src/bitmap/iter.rs:143:41
    |
143 |     fn from_iter<I: IntoIterator<Item = &u32>>(iterator: I) -> RoaringBitmap {
    |                                         ^ explicit lifetime name needed here
```

**Review by Kerollmops — APPROVED** [review]
Thank you!
bors merge

**Pull request merged**